Project-Aurora/Project-Aurora/Devices/YeeLight/YeeLightDevice.cs

```diff
@@ -0,0 +1,192 @@
+﻿using Aurora.Settings;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Drawing;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Aurora.Devices.YeeLight
+{
+    public class YeeLightDevice : Device
+    {
+        private string devicename = "YeeLight";
+
+        private bool isConnected;
+        private Stopwatch sw = new Stopwatch();
+
+        private Stopwatch watch = new Stopwatch();
+        private YeeLight light = new YeeLight();
+        private long lastUpdateTime = 0;
+
+        private VariableRegistry default_registry = null;
+
+        public string GetDeviceDetails()
+        {
+            if (isConnected)
+                return devicename + ": Connected";
+            else
+                return devicename + ": Not connected";
+        }
+
+        public string GetDeviceName()
+        {
+            return devicename;
+        }
+
+        public bool Initialize()
+        {
+            if (!isConnected)
+            {
+                try
+                {
+                    Connect();
+                }
+                catch (Exception exc)
+                {
+                    Global.logger.Error($"Device {devicename} encountered an error during Connecting. Exception: {exc}");
+                    isConnected = false;
+
+                    return false;
+                }
+            }
+
+            return isConnected;
+        }
+
+        public bool IsConnected()
+        {
+            return isConnected;
+        }
+
+        public bool IsInitialized()
+        {
+            return IsConnected();
+        }
+
+        public bool IsKeyboardConnected()
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool IsPeripheralConnected()
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool Reconnect()
+        {
+            light.CloseConnection();
+
+            isConnected = false;
+
+            Connect();
+            return true;
+        }
+
+        public void Reset()
+        {
+            Reconnect();
+        }
+
+        public void Shutdown()
+        {
+            light.CloseConnection();
+
+            isConnected = false;
+
+            if (sw.IsRunning)
+                sw.Stop();
+        }
+
+        public void Connect(DoWorkEventArgs token = null)
+        {
+            try
+            {
+                if (!light.isConnected())
+                {
+                    IPAddress localIP;
+                    IPAddress.TryParse(Global.Configuration.VarRegistry.GetVariable<string>($"{devicename}_IP"), out IPAddress lightIP);
```

Comment:
This `TryParse` should be handled, if it's not parsed correctly then it'll try and connect when it really shouldn't? Could throw up a `MessageBox` if you make sure that it is a once off. Or it might make sense for the Variable editor to have an ability to have a regex for input validity checking? What do you think?